### PR TITLE
P#52017  fix php di issue and type errors

### PR DESCRIPTION
**Pull request by vu-vuong-esg**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
     "config": {
         "platform": {
-            "php": "7.0.33"
+            "php": "7.2"
         }
     },
 	"require-dev": {
-		"phpunit/phpunit": "6 - 7",
+		"phpunit/phpunit": "6.5.14",
```
Comment (jayay) [inline composer.json:8]: Thank you. Do we need to make a new project to update PHPUnit? Theoretically, PHPUnit 8.x supports PHP 7.2, but it looks like v7.x is preferred. Did you happen to try v7.x or v8.x?

Comment (vu-vuong-esg) [reply composer.json]: I updated to PHPunit 7 and fix the errors. Can you take a look

File: tests/TestClassDatabaseChanges.php [hunk 2]
```diff
@@ -138,9 +138,9 @@ public function testInstallMigrationsDataSimilarEstates(): array
 		$this->assertEquals('Field 1', $newFields[0]);
 		$this->assertEquals('Field 2', $newFields[1]);
 		$this->assertEquals('Field 3', $newFields[2]);
-		$this->assertTrue(true, $newRadius);
```
Comment (jayay) [inline tests/TestClassDatabaseChanges.php:141]: is `assertTrue` correct here? Looks like it should be `assertEquals`?

**Review by jayay — CHANGES_REQUESTED** [review]

File: tests/TestClassDatabaseChanges.php (hunk 2 above)
Comment (vu-vuong-esg) [reply tests/TestClassDatabaseChanges.php:141]: ok let me check it tomorrow

File: tests/TestClassDatabaseChanges.php [hunk 3]
```diff
@@ -138,7 +138,7 @@ public function testInstallMigrationsDataSimilarEstates(): array
 		$this->assertEquals('Field 1', $newFields[0]);
 		$this->assertEquals('Field 2', $newFields[1]);
 		$this->assertEquals('Field 3', $newFields[2]);
-		$this->assertTrue(true, (string)$newRadius);
+		$this->assertEquals(35, $newRadius);
 		$this->assertTrue(true, (string)$newSameKind);
```
Comment (jayay) [inline tests/TestClassDatabaseChanges.php]: there are two more of these

Comment (vu-vuong-esg) [reply tests/TestClassDatabaseChanges.php]: I updated it

Comment (jayay) [reply tests/TestClassDatabaseChanges.php]: What exactly is the `(string)` cast there for? If the usage of `assertTrue` was intended, the first parameter would have to be removed (`true`).

Comment (vu-vuong-esg) [reply tests/TestClassDatabaseChanges.php]: ok I will update it

**Review by jayay — APPROVED** [review]

**Comment by jayay** [comment]
Would you please also set the minimum required PHP version in `readme.txt`?

https://github.com/onOfficeGmbH/oo-wp-plugin/blob/8d290b7bce18a2ae1e9a59a449e168079f3ab57e/readme.txt#L6

**Review by jayay — DISMISSED** [review]

**Comment by michaelarnold-dev** [comment]
@hungnc89 Could you plrease resolve the conflicts?

**Comment by tang-hien-egs** [comment]
@michaelarnold-dev , i fixed it, can you check it?

**Comment by jayay** [comment]
Sadly, this will always conflict until merged.

**Comment by nglelinh** [comment]
@jayay sorry we do not get your last comment. Do you need any further action on outside in this PR?

**Comment by jayay** [comment]
@nglelinh everything is fine. I just wanted to imply that pretty much every change to composer.lock will lead to new conflicts.

**Review by jmaas-onoffice — APPROVED** [review]

**Review by jmaas-onoffice — APPROVED** [review]

**Pull request merged**